### PR TITLE
Track progress of pod and container deployment

### PR DESCRIPTION
**Pull request by bormanp**
Deploy() will not fail if one of the containers fails to deploy due to ErrImagePull.
An optional --progress flag can be passed to kne deploy to cause verbose progress to be displayed.

### CODE REVIEW

File: deploy/status.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
+package deploy
```
[the remark this answers is not in the document]
Comment (bormanp) [reply deploy/status.go]: Done

File: deploy/deploy.go [hunk 2]
```diff
@@ -185,6 +189,17 @@ func (d *Deployment) Deploy(ctx context.Context, kubecfg string) error {
 	}
 	log.V(1).Info("Found k8s versions:\n", output)
 
+	ctx, cancel := context.WithCancel(ctx)
+
+	// Watch the containter status of the pods so we can fail if a container fails to start running.
+	if w, _ := NewWatcher(ctx, kClient, cancel); w != nil {
```
[the remark this answers is not in the document]
Comment (bormanp) [reply deploy/deploy.go]: Done

**Review by alexmasi — APPROVED** [review]

**Pull request merged**